### PR TITLE
wofi: add module

### PR DESCRIPTION
**Pull request by christoph-heiss**
### Description

Adds [wofi](https://hg.sr.ht/~scoopta/wofi) (a launcher/menu program for wlroots based wayland compositors such as sway) as a new module.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by christoph-heiss** [comment]
Updated this a bit by
- Removing myself from `modules/lib/maintainers.nix`, as I'm listed in the nixpkgs' list now
- Simplified config generation a lot

Happy to change things as needed to get this merged.

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Comment by christoph-heiss** [comment]
Thanks for the review @sumnerevans, I've changed all the things according to your suggestions!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -580,6 +580,9 @@ Makefile                                              @thiagokokada
 /modules/targets/darwin                               @midchildan
 /tests/modules/targets-darwin                         @midchildan
 
+/modules/programs/wofi.nix                            @christoph-heiss
+/tests/modules/programs/wofi                          @christoph-heiss
+
```
Comment (sumnerevans) [inline .github/CODEOWNERS:585]: Oh, just realized this is a program not a service. It should go up with the programs around line 350

File: tests/default.nix [hunk 2]
```diff
@@ -165,6 +165,7 @@ import nmt {
     ./modules/programs/terminator
     ./modules/programs/thunderbird
     ./modules/programs/waybar
+    ./modules/programs/wofi
```
Comment (sumnerevans) [inline tests/default.nix:168]: should come after wlogout

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Comment by christoph-heiss** [comment]
Thanks again, fixed both these things!

**Review by sumnerevans — APPROVED** [review]

**Pull request merged**